src/main/java/io/tus/java/client/TusUploader.java

```diff
@@ -36,16 +38,39 @@ public class TusUploader {
      * Begin a new upload request by opening a PATCH request to specified upload URL. After this
      * method returns a connection will be ready and you can upload chunks of the file.
      *
+     * @deprecated This constructor is deprecated in favor of the new one that includes the option
+     * to add a proxy.
+     *
      * @param client Used for preparing a request ({@link TusClient#prepareConnection(HttpURLConnection)}
      * @param upload {@link TusUpload} to be uploaded.
      * @param uploadURL URL to send the request to
      * @param input Stream to read (and seek) from and upload to the remote server
      * @param offset Offset to read from
      * @throws IOException Thrown if an exception occurs while issuing the HTTP request.
      */
+    @Deprecated
     public TusUploader(TusClient client, TusUpload upload, URL uploadURL, TusInputStream input, long offset)
-            throws IOException {
+        throws IOException {
+        this(client, upload, uploadURL, Proxy.NO_PROXY, input, offset);
+    }
+
+    /**
+     * Begin a new upload request by opening a PATCH request to specified upload URL. After this
+     * method returns a connection will be ready and you can upload chunks of the file.
+     *
+     * @param client Used for preparing a request ({@link TusClient#prepareConnection(HttpURLConnection)}
+     * @param upload {@link TusUpload} to be uploaded.
+     * @param uploadURL URL to send the request to
+     * @param uploadProxy Proxy to use with the {@link #uploadURL}
+     * @param input Stream to read (and seek) from and upload to the remote server
+     * @param offset Offset to read from
+     * @throws IOException Thrown if an exception occurs while issuing the HTTP request.
+     */
+    public TusUploader(
+        TusClient client, TusUpload upload, URL uploadURL, Proxy uploadProxy, TusInputStream input, long offset
```

Comment:
Let's not add another constructor and instead add a method `setProxy` that can optionally be called. This way we do not have to add a new constructor and deprecate the current one.

Reply:
Sure!